### PR TITLE
feat(webhooks): [nan-1909] add in webhook support for airtable

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
Add in webhook support for airtable using a catch all webhook subscription and resolving the webhook id via the metadata. Relates to https://github.com/NangoHQ/integration-templates/pull/68

## Issue ticket number and link
NAN-1909

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/server/lib/webhook/internal-nango.ts [hunk 1]
```diff
@@ -56,6 +56,14 @@ export const internalNango: InternalNango = {
...
                 connections = [connection];
             }
+        } else if (propName && propName.includes('metadata.')) {
+            const strippedMetadata = propName.replace('metadata.', '');
+            connections = await connectionService.findConnectionsByMetadataValue({
+                metadataProperty: strippedMetadata,
+                payloadIdentifier: get(body, connectionIdentifier),
+                configId: integration.id,
+                environmentId: integration.environment_id
+            });
```
Comment (TBonnin) [inline packages/server/lib/webhook/internal-nango.ts:66]: can you show an example of a webhook body sent by Airtable. I am having a hard time visualizing what the metadata property is? thank you

Comment (khaliqgant) [reply packages/server/lib/webhook/internal-nango.ts:66]: Webhook body:
```
{
  "base": {
    "id": "appaYNlX7K9HdmkDr"
  },
  "webhook": {
    "id": "achQfrhelikFY4IKb"
  },
  "timestamp": "2024-10-22T13:18:24.929Z"
}
```

Metadata stucture:
```
{
    "webhooks": {
        "achAI0q44ruSk65rS": "dJfr/OW2JWkZLNvJsdjefshlPvvRBT3Hk5bPxNNj6NVOm9OGWdcrLhFdd8ja54Eq0t6/zSrflNnPMYT/w84xX2/aSwNhHc2N78Q1Jy9DE/qwbq2mRIY9Eih0EuhZvDjIrAUwaFrm7GOod5K6/dO8fSNcUwnaojGIyS+R/pwgaUw=",
        "achQfrhelikFY4IKb": "faPQvmagaFHopsrBzmy2QdHR9vk0sv8VaKsTVQq21b5F4+z8zm4ChcAlTmJWQkK7NOA1HhhtRrNjSM9v8tQ01YEJgb2FzAdZOFmvFQ+mM6nKo8v1GQEBq0PTTJpu+g3YQ4BpaQqPKl9r9dhtWhH6nuITldQtC9+CkGn/xv734YY="
    }
}
```

File: packages/shared/lib/services/connection.service.ts [hunk 2]
```diff
@@ -849,6 +849,36 @@ class ConnectionService {
...
+        if (!configId) {
+            return null;
+        }
+
+        const result = await db.knex
+            .from<StoredConnection>(`_nango_connections`)
+            .select('*')
+            .where({ environment_id: environmentId, config_id: configId })
+            // escape the question mark so it doesn't try to bind it as a parameter
+            .where(db.knex.raw(`metadata->? \\? ?`, [metadataProperty, payloadIdentifier]))
```
Comment (TBonnin) [inline packages/shared/lib/services/connection.service.ts:872]: you can use `.whereRaw` instead of `.where(db.knex.raw...)`

**Review by TBonnin — APPROVED** [review]

File: packages/shared/lib/services/connection.service.ts (hunk 2 above)
Comment (khaliqgant) [reply packages/shared/lib/services/connection.service.ts:872]: I tried this at first and I think I ran into issues with the parameters and the ? with encoding

**Pull request merged**